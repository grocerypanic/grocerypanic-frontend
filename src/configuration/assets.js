const Assets = {
  favicon: `${process.env.PUBLIC_URL}/favicon.ico`,
  nonBreakingSpace: "\u00A0",
  cart: `${process.env.PUBLIC_URL}/images/cart_desktop.jpg`,
  bill: `${process.env.PUBLIC_URL}/images/bill_desktop.jpg`,
  cart_mobile: `${process.env.PUBLIC_URL}/images/cart_mobile.jpg`,
  bill_mobile: `${process.env.PUBLIC_URL}/images/bill_mobile.jpg`,
};

export default Assets;
