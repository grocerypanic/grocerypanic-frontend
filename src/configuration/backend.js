export const SafeMethods = ["GET", "HEAD", "OPTIONS", "TRACE"];
export const ItemFilters = ["preferred_stores", "shelf"];
export const FilterTag = "filter_tag";

export const Constants = {
  csrfLocalStorage: "csrf-token",
  csrfHeaderName: "X-CSRFToken",
  genericAPIError: "API Error",
  csrfErrorMessage: "Refresh csrf and try again.",
  alreadyRegistered: "User is already registered with this e-mail address.",
  minimumQuanity: 0,
  maximumQuantity: 10000,
  minimumShelfLife: 1,
  maximumShelfLife: 365 * 3,
  defaultShelfLife: 7,
  minimumPrice: 0,
  maximumPrice: 9999,
  maximumTransactionHistory: 14,
  AnalyticsCookieName: "PanicDataConsent-" + process.env.NODE_ENV.toUpperCase(),
  duplicateObjectApiError: { non_field_errors: ["Object already exists."] },
  pageLookupParam: "page",
  pageOverrideParam: "all_results",
};

export const Providers = {
  google: "google",
  facebook: "facebook",
};

export const Paths = {
  googleLogin: "/api/v1/auth/social/google/",
  facebookLogin: "/api/v1/auth/social/facebook/",
  manageShelves: "/api/v1/shelves/",
  manageStores: "/api/v1/stores/",
  manageItems: "/api/v1/items/",
  refreshCSRF: "/api/v1/auth/csrf/",
  manageTransactions: "/api/v1/transactions/",
};

const month = 30;
const year = 365;

export const ShelfLifeConstants = [
  { name: "3 Days", id: 3 },
  { name: "1 Week", id: 7 },
  { name: "2 Weeks", id: 14 },
  { name: "3 Weeks", id: 21 },
  { name: "1 Month", id: month },
  { name: "2 Months", id: month * 2 },
  { name: "3 Months", id: month * 3 },
  { name: "6 Months", id: month * 6 },
  { name: "1 Year", id: year },
  { name: "2 Years", id: year * 2 },
  { name: "3 Years", id: year * 3 },
];
