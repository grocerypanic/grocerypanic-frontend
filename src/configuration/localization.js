import i18n from "i18next";
import LanguageDetector from "i18next-browser-languagedetector";
import { Strings } from "./strings";

const i18nConfig = {
  fallbackLng: {
    default: ["en"],
  },
  debug: true,
  ns: ["translation"],
  defaultNS: "translation",
  keySeparator: ".",
  interpolation: {
    escapeValue: false,
    formatSeparator: ",",
  },
  react: {
    wait: true,
  },
};

i18n.use(LanguageDetector).init({
  resources: {
    ...Strings,
  },
  ...i18nConfig,
});

export default i18n;
