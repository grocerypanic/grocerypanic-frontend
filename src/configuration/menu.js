import Routes from "./routes";

const Options = [
  { name: "All Items", location: Routes.items },
  { name: "Shelves", location: Routes.shelves },
  { name: "Stores", location: Routes.stores },
];

export default Options;
