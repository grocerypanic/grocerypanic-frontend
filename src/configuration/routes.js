const Routes = {
  root: "/",
  menu: "/menu",
  about: "/about",
  privacy: "/privacy",
  signin: "/signin",
  shelves: "/shelves",
  splash: "/",
  stores: "/stores",
  items: "/items",
  create: "/create",
  details: "/details/:id",
  goBack: "back",
};

export const External = {
  feedback: "https://www.sharedvisionsolutions.com/#contact",
  credit: "https://www.sharedvisionsolutions.com/",
};

export default Routes;
