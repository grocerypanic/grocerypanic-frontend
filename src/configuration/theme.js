export const primary = "#006600";
export const white = "#FFFFFF";
export const secondary = "#868e96";
export const tertiary = "#808080";
export const error = "#FF0000";
export const regular = "#000000";
export const success = "#5cb85c";
export const highlight = "blue";
export const selected = "red";
export const danger = "#d9534f";
export const dark = "#495057";
export const black = "#000000";
export const warning = "#ffc107";

export const ui = {
  alertTimeout: 2000,
  resizeTimeout: 1,
};

export const navbarSize = "50px";
export const tabSize = "26px";
export const itemAttributes = {
  height: 55,
  padding: 5,
  border: 10,
  spacing: 5,
  innerHeight: 40,
  placeHolderPadding: 12,
};

export const headerMobileThreshold = 480;
export const mobileThreshold = 480;

export const graph = {
  quantityLine: "rgb(2, 117, 216)",
  changeLine: "rgb(92, 184, 92)",
  topPadding: -12,
  holdingPatternHeight: "134px",
  chartContainerHeight: "150px",
  holdingPatternColor: "secondary",
  holdingPatternScale: 1,
  holdingPatternDivHeight: 20,
  holdingPatternAnimation: "grow",
};
