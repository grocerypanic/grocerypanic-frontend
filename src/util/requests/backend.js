import { Constants, SafeMethods } from "../../configuration/backend";
import debug from "../debug";

const Backend = (method, path, data = null) => {
  // Prepend Backend Hostname and Protocol If Needed
  let url;
  if (!path.includes(process.env.REACT_APP_PANIC_BACKEND)) {
    url = `${process.env.REACT_APP_PANIC_BACKEND}${path}`;
  } else {
    url = path;
  }

  debug(`API ${method}:\n ${url}`);
  if (data) debug(`Body:\n ${JSON.stringify(data)}`);
  let csrf;
  let statusCode;

  const options = {
    method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    credentials: "include",
  };

  if (data) options.body = JSON.stringify(data);
  if (!SafeMethods.includes(method)) {
    csrf = localStorage.getItem(Constants.csrfLocalStorage);
    if (csrf) options.headers[Constants.csrfHeaderName] = csrf;
  }

  const response = fetch(url, options)
    .then(async (fetchResponse) => {
      const contentType = fetchResponse.headers.get("content-type");
      statusCode = fetchResponse.status;
      if (contentType === null) return null;
      if (contentType.startsWith("application/json"))
        return await fetchResponse.json();
      return fetchResponse.text();
    })
    .then((fetchResponseDecoded) => {
      debug(`API Response Status Code:\n ${statusCode}`);
      if (fetchResponseDecoded) {
        debug(`API Response Data:\n`);
        debug(fetchResponseDecoded);
      }
      return [fetchResponseDecoded, statusCode];
    })
    .catch((err) => {
      return [Constants.genericAPIError, 500];
    });
  return response;
};

export default Backend;
