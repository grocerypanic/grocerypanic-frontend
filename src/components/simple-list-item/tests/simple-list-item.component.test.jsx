import React from "react";
import { render, cleanup, waitFor, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Router, MemoryRouter } from "react-router-dom";
import { createBrowserHistory } from "history";

import SimpleListItem from "../simple-list-item.component";

import Strings from "../../../configuration/strings";
import Routes from "../../../configuration/routes";
import { FilterTag } from "../../../configuration/backend";

jest.mock("../../../configuration/theme", () => {
  return {
    __esModule: true,
    ...jest.requireActual("../../../configuration/theme"),
    ui: { alertTimeout: 10 },
  };
});

jest.mock("../../../configuration/theme");

const mockAdd = jest.fn();
const mockDelete = jest.fn();
const setSelected = jest.fn();
const setErrorMsg = jest.fn();
const setCreated = jest.fn();
const setLongPress = jest.fn();
const setActionMsg = jest.fn();

// Mock Api Data
const mockData = [
  { id: 1, name: "Shelf1" },
  { id: 2, name: "Shelf2" },
  { id: 3, name: "Shelf3" },
];

// Test Data Defaults

let test = {
  item: mockData[0],
  allItems: mockData,
  listFunctions: {
    add: mockAdd,
    del: mockDelete,
    setSelected: setSelected,
    setErrorMsg: setErrorMsg,
    setCreated: setCreated,
    setLongPress: setLongPress,
    setActionMsg: setActionMsg,
  },
  listValues: {
    transaction: false,
    selected: null,
    errorMsg: null,
    longPress: false,
  },
  redirectTag: "redirectTag",
};

let utils;
let current;
describe("Setup Environment", () => {
  beforeEach(() => {
    current = test;
  });
  describe("when not rendering a newly created item", () => {
    beforeEach(() => {
      current.item = mockData[0];
    });
    describe("when outside of a transaction", () => {
      beforeEach(() => {
        current.listValues.transaction = false;
      });
      describe("when not selected", () => {
        beforeEach(() => {
          current.listValues.selected = null;
          jest.clearAllMocks();
          utils = render(
            <MemoryRouter>
              <SimpleListItem {...current} />
            </MemoryRouter>
          );
        });
        afterEach(cleanup);

        it("renders only the expected components", () => {
          expect(current.listValues.selected).not.toBe(current.item.id);
          expect(current.listValues.transaction).toBeFalsy();
          expect(utils.queryByTestId("listElement")).toBeTruthy();
          expect(utils.queryByTestId("deleteButton")).toBeFalsy();
          expect(utils.queryByTestId("inputElement")).toBeFalsy();
          expect(utils.queryByTestId("saveButton")).toBeFalsy();
        });

        it("handles a short click on items as expected", async (done) => {
          const itemComponent = utils.queryByTestId("listElement");
          fireEvent.mouseDown(itemComponent, "click");
          fireEvent.mouseUp(itemComponent, "click");
          await waitFor(() => expect(setCreated).toBeCalledWith(false));
          expect(setSelected).toBeCalledWith(current.item.id);
          done();
        });

        it("handles a click on item names as expected (by doing nothing)", async (done) => {
          const itemComponent = utils.getByTestId("ListTitle");
          fireEvent.click(itemComponent, "click");
          done();
        });
      });

      describe("when selected, and longPress is not marked", () => {
        beforeEach(() => {
          current.listValues.selected = current.item.id;
          current.listValues.longPress = false;
          jest.clearAllMocks();
          const history = createBrowserHistory({ basename: "/" });
          utils = render(
            <Router history={history}>
              <SimpleListItem {...current} />
            </Router>
          );
        });
        afterEach(cleanup);

        it("handles a short click on items as expected", async (done) => {
          const itemComponent = utils.queryByTestId("listElement");
          fireEvent.mouseDown(itemComponent, "click");
          fireEvent.mouseUp(itemComponent, "click");
          setTimeout(() => fireEvent.mouseUp(itemComponent, "click"), 500);
          await waitFor(() => expect(setCreated).toBeCalledTimes(0));
          expect(setSelected).toBeCalledWith(current.item.id);
          done();
        });

        it("handles a long click on items as expected", async (done) => {
          const itemComponent = utils.queryByTestId("listElement");
          fireEvent.mouseDown(itemComponent, "click");
          setTimeout(() => fireEvent.mouseUp(itemComponent, "click"), 500);
          await waitFor(() => expect(setLongPress).toBeCalledTimes(1));
          done();
        });

        it("handles a click on item names as expected (navigate)", () => {
          const itemComponent = utils.getByTestId("ListTitle");
          expect(window.location.pathname).not.toBe(Routes.items);
          fireEvent.click(itemComponent, "click");
          expect(window.location.pathname).toBe(Routes.items);
          expect(window.location.search).toBe(
            `?${FilterTag}=${current.item.name}&redirectTag=${current.item.id}`
          );
        });

        it("renders only the expected components", () => {
          expect(current.listValues.selected).toBe(current.item.id);
          expect(current.listValues.transaction).toBeFalsy();
          expect(utils.queryByTestId("listElement")).toBeTruthy();
          expect(utils.queryByTestId("deleteButton")).toBeFalsy();
          expect(utils.queryByTestId("inputElement")).toBeFalsy();
          expect(utils.queryByTestId("saveButton")).toBeFalsy();
        });
      });

      describe("when selected, and longPress is marked", () => {
        beforeEach(() => {
          current.listValues.selected = current.item.id;
          current.listValues.longPress = true;
          jest.clearAllMocks();
          utils = render(
            <MemoryRouter>
              <SimpleListItem {...current} />
            </MemoryRouter>
          );
        });
        afterEach(cleanup);

        it("renders only the expected components", () => {
          expect(current.listValues.selected).toBe(current.item.id);
          expect(current.listValues.transaction).toBeFalsy();
          expect(utils.queryByTestId("listElement")).toBeTruthy();
          expect(utils.queryByTestId("deleteButton")).toBeTruthy();
          expect(utils.getByText(Strings.SimpleList.DeleteButton)).toBeTruthy();
          expect(utils.queryByTestId("inputElement")).toBeFalsy();
          expect(utils.queryByTestId("saveButton")).toBeFalsy();
        });

        it("calls handleDelete when delete is pressed", async (done) => {
          const deleteButton = utils.queryByTestId("deleteButton");
          fireEvent.click(deleteButton, "click");
          expect(setSelected).toBeCalledWith(null);
          expect(mockDelete).toBeCalledWith(current.item.id);
          await (() => expect(setActionMsg).toBeCalledTimes(2));
          expect(setActionMsg).toBeCalledWith(
            `${Strings.SimpleList.DeletedAction} ${current.item.name}`
          );
          done();
        });

        it("handles a click on item names as expected (by doing nothing)", () => {
          const itemComponent = utils.getByTestId("ListTitle");
          fireEvent.click(itemComponent, "click");
          expect(setActionMsg).toBeCalledTimes(0);
        });
      });
    });
    describe("when inside of a transaction", () => {
      beforeEach(() => {
        current.listValues.transaction = true;
      });
      describe("when not selected", () => {
        beforeEach(() => {
          current.listValues.selected = null;
          jest.clearAllMocks();
          utils = render(
            <MemoryRouter>
              <SimpleListItem {...current} />
            </MemoryRouter>
          );
        });
        afterEach(cleanup);

        it("renders only the expected components", () => {
          expect(current.listValues.selected).not.toBe(current.item.id);
          expect(current.listValues.transaction).toBeTruthy();
          expect(utils.queryByTestId("listElement")).toBeTruthy();
          expect(utils.queryByTestId("deleteButton")).toBeFalsy();
          expect(utils.queryByTestId("inputElement")).toBeFalsy();
        });

        it("handles a click on items as expected (by doing nothing)", () => {
          const itemComponent = utils.queryByTestId("listElement");
          fireEvent.click(itemComponent, "click");
          expect(setCreated).toBeCalledTimes(0);
          expect(setSelected).toBeCalledTimes(0);
        });
      });

      describe("when selected", () => {
        beforeEach(() => {
          current.listValues.selected = current.item.id;
          jest.clearAllMocks();
          utils = render(
            <MemoryRouter>
              <SimpleListItem {...current} />
            </MemoryRouter>
          );
        });
        afterEach(cleanup);

        it("handles a click on item names as expected (by doing nothing)", () => {
          const itemComponent = utils.getByTestId("ListTitle");
          fireEvent.click(itemComponent, "click");
        });

        it("renders only the expected components, without the delete button", () => {
          expect(current.listValues.selected).toBe(current.item.id);
          expect(current.listValues.transaction).toBeTruthy();
          expect(utils.queryByTestId("listElement")).toBeTruthy();
          expect(utils.queryByTestId("deleteButton")).toBeFalsy();
          expect(utils.queryByTestId("inputElement")).toBeFalsy();
        });
      });
    });
  });

  describe("when rendering a newly created item", () => {
    beforeEach(() => {
      current.item.id = -1;
      current.item.name = "";
    });
    describe("when outside of a transaction", () => {
      beforeEach(() => {
        current.listValues.transaction = false;
      });
      describe("when there is an error message", () => {
        beforeEach(() => {
          current.listValues.errorMsg = "Some Existing Error Message";
          current.listValues.selected = null;

          jest.clearAllMocks();
          utils = render(
            <MemoryRouter>
              <SimpleListItem {...current} />
            </MemoryRouter>
          );
        });
        afterEach(cleanup);

        it("renders only the expected components, including the save button", () => {
          expect(current.listValues.selected).not.toBe(current.item.id);
          expect(current.listValues.transaction).toBeFalsy();
          expect(utils.queryByTestId("listElement")).toBeTruthy();
          expect(utils.queryByTestId("deleteButton")).toBeFalsy();
          expect(utils.queryByTestId("inputElement")).toBeTruthy();
          expect(utils.queryByTestId("saveButton")).toBeTruthy();
          expect(utils.getByText(Strings.SimpleList.SaveButton)).toBeTruthy();
          expect(setErrorMsg).toBeCalledTimes(0);
        });

        it("handles input as expected", async (done) => {
          const input = utils.queryByTestId("inputElement");
          await userEvent.type(input, "Hello, World!");
          expect(setErrorMsg).toBeCalledWith(null);
          done();
        });
      });
      describe("when there is no error message", () => {
        beforeEach(() => {
          current.listValues.errorMsg = "";
          current.listValues.selected = null;

          jest.clearAllMocks();
          utils = render(
            <MemoryRouter>
              <SimpleListItem {...current} />
            </MemoryRouter>
          );
        });
        afterEach(cleanup);

        it("handles input as expected", async (done) => {
          const input = utils.queryByTestId("inputElement");
          await userEvent.type(input, "Hello, World!");
          expect(setErrorMsg).toBeCalledTimes(0);
          done();
        });

        it("handles submit without doing anything", async (done) => {
          const input = utils.queryByTestId("inputElement");
          await userEvent.type(input, "Hello, World!");
          fireEvent.submit(input);
          expect(setErrorMsg).toBeCalledTimes(0);
          await (() => expect(setActionMsg).toBeCalledTimes(0));
          done();
        });

        it("handles saveButton click as expected", async (done) => {
          const input = utils.queryByTestId("inputElement");
          const save = utils.queryByTestId("saveButton");
          await userEvent.type(input, "Hello, World!");
          fireEvent.click(save);
          expect(setErrorMsg).toBeCalledTimes(0);
          await (() => expect(setActionMsg).toBeCalledTimes(2));
          expect(setActionMsg).toBeCalledWith(
            `${Strings.SimpleList.CreatedAction} Hello, World!`
          );
          done();
        });

        it("handles submit as expected with invalid input", async (done) => {
          const input = utils.queryByTestId("inputElement");
          await userEvent.type(input, "a");
          const save = utils.queryByTestId("saveButton");
          fireEvent.click(save);
          expect(setErrorMsg).toBeCalledWith(
            Strings.SimpleList.ValidationFailure
          );
          await (() => expect(setActionMsg).toBeCalledTimes(0));
          done();
        });

        it("handles submit as expected with existing input", async (done) => {
          const input = utils.queryByTestId("inputElement");
          await userEvent.type(input, "Shelf2");
          const save = utils.queryByTestId("saveButton");
          fireEvent.click(save);
          expect(setErrorMsg).toBeCalledWith(
            Strings.SimpleList.ValidationAlreadyExists
          );
          await (() => expect(setActionMsg).toBeCalledTimes(0));
          done();
        });
      });
    });

    describe("when inside of a transaction", () => {
      beforeEach(() => {
        current.listValues.transaction = true;
        current.listValues.selected = null;
        jest.clearAllMocks();
        utils = render(
          <MemoryRouter>
            <SimpleListItem {...current} />
          </MemoryRouter>
        );
      });

      afterEach(cleanup);

      it("renders only the expected components, without the save button", () => {
        expect(current.listValues.selected).not.toBe(current.item.id);
        expect(current.listValues.transaction).toBeTruthy();
        expect(utils.queryByTestId("listElement")).toBeTruthy();
        expect(utils.queryByTestId("deleteButton")).toBeFalsy();
        expect(utils.queryByTestId("inputElement")).toBeTruthy();
        expect(utils.queryByTestId("saveButton")).toBeFalsy();
      });

      it("handles a click on items as expected (by doing nothing)", () => {
        const itemComponent = utils.queryByTestId("listElement");
        fireEvent.click(itemComponent, "click");
        expect(setCreated).toBeCalledTimes(0);
        expect(setSelected).toBeCalledTimes(0);
      });

      it("handles a long click on items as expected", async (done) => {
        const itemComponent = utils.queryByTestId("listElement");
        fireEvent.mouseDown(itemComponent, "click");
        setTimeout(() => {
          fireEvent.mouseUp(itemComponent, "click");
          expect(setLongPress).toBeCalledTimes(0);
          done();
        }, 500);
      });
    });
  });
});
