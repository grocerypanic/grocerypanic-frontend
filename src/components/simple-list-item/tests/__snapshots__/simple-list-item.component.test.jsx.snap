// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup Environment when not rendering a newly created item when inside of a transaction when not selected should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:last-child {
  margin-left: auto;
}

.c0 .simple-list-item-title-div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
}

<li
  class="c0"
  data-testid="listElement"
>
  <div
    class="c1 simple-list-item-title-div"
    data-testid="ListTitle"
  >
    Shelf1
  </div>
  <div />
</li>
`;

exports[`Setup Environment when not rendering a newly created item when inside of a transaction when selected should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:last-child {
  margin-left: auto;
}

.c0 .simple-list-item-title-div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
}

<li
  class="c0"
  data-testid="listElement"
>
  <div
    class="c1 simple-list-item-title-div"
    data-testid="ListTitle"
  >
    Shelf1
  </div>
  <div />
</li>
`;

exports[`Setup Environment when not rendering a newly created item when outside of a transaction when not selected should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:last-child {
  margin-left: auto;
}

.c0 .simple-list-item-title-div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
}

<li
  class="c0"
  data-testid="listElement"
>
  <div
    class="c1 simple-list-item-title-div"
    data-testid="ListTitle"
  >
    Shelf1
  </div>
  <div />
</li>
`;

exports[`Setup Environment when not rendering a newly created item when outside of a transaction when selected, and longPress is marked should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:last-child {
  margin-left: auto;
}

.c0 .simple-list-item-title-div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
}

<li
  class="c0"
  data-testid="listElement"
>
  <div
    class="c1 simple-list-item-title-div"
    data-testid="ListTitle"
  >
    Shelf1
  </div>
  <div>
    <button
      class="btn btn-danger"
      data-testid="deleteButton"
      style="height: 40px;"
    >
      delete
    </button>
  </div>
</li>
`;

exports[`Setup Environment when not rendering a newly created item when outside of a transaction when selected, and longPress is not marked should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:last-child {
  margin-left: auto;
}

.c0 .simple-list-item-title-div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
}

<li
  class="c0"
  data-testid="listElement"
>
  <div
    class="c1 simple-list-item-title-div"
    data-testid="ListTitle"
  >
    Shelf1
  </div>
  <div />
</li>
`;

exports[`Setup Environment when rendering a newly created item when inside of a transaction should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:last-child {
  margin-left: auto;
}

.c0 .simple-list-item-title-div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
}

<form>
  <li
    class="c0"
    data-testid="listElement"
  >
    <div
      class="c1"
    >
      <input
        data-testid="inputElement"
        id="newItem"
        name="newItem"
        readonly=""
        required=""
        size="15"
        type="newItem"
        value=""
      />
    </div>
    <div />
  </li>
</form>
`;

exports[`Setup Environment when rendering a newly created item when outside of a transaction when there is an error message should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:last-child {
  margin-left: auto;
}

.c0 .simple-list-item-title-div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
}

<form>
  <li
    class="c0"
    data-testid="listElement"
  >
    <div
      class="c1"
    >
      <input
        data-testid="inputElement"
        id="newItem"
        name="newItem"
        required=""
        size="15"
        type="newItem"
        value=""
      />
    </div>
    <div>
      <button
        class="btn btn-success"
        data-testid="saveButton"
        style="height: 40px;"
      >
        <span>
          save
        </span>
      </button>
    </div>
  </li>
</form>
`;

exports[`Setup Environment when rendering a newly created item when outside of a transaction when there is no error message should match the snapshot on file (styles) 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background: #FFFFFF;
  margin-bottom: 5px;
  padding: 5px;
  color: blue;
  min-width: 220px;
}

.c0 div:last-child {
  margin-left: auto;
}

.c0 .simple-list-item-title-div:hover {
  color: red;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 40px;
}

<form>
  <li
    class="c0"
    data-testid="listElement"
  >
    <div
      class="c1"
    >
      <input
        data-testid="inputElement"
        id="newItem"
        name="newItem"
        required=""
        size="15"
        type="newItem"
        value=""
      />
    </div>
    <div>
      <button
        class="btn btn-success"
        data-testid="saveButton"
        style="height: 40px;"
      >
        <span>
          save
        </span>
      </button>
    </div>
  </li>
</form>
`;
